### PR TITLE
Lock a lockfile to make sure the judgedaemon is only running once.

### LINKED ISSUE #1776: Add locking for judgedaemon to prevent duplicate instances
When you accidentally start a duplicate instance of the same judgedaemon (same host and same CPU core), then weird errors happen.

We should add some locking that detects and prevents this from happening. During the WFs in Dhaka this happened at least to me, @meisterT and @nickygerritsen, since there was already a judgedaemon running as a service.

### PR DESCRIPTION
**Pull request by nickygerritsen**
Fixes #1776

The logfile is unique per daemon, so it seemed like a good file to use?

### CODE REVIEW

File: judge/judgedaemon.main.php [hunk 1]
```diff
@@ -538,6 +538,14 @@ function fetch_executable_internal(
     exit(1);
 }
 
+if (defined('STDLOG')) {
+    // Add a lock to the logfile; this makes sure we can only run the daemon once
+    if (!flock(STDLOG, LOCK_EX | LOCK_NB)) {
+        echo "Can lock logfile, is a daemon already running?\n";
```
Comment (vmcj) [inline judge/judgedaemon.main.php:544]: If you can, should you exit?

Comment (nickygerritsen) [reply judge/judgedaemon.main.php:544]: Can't*. But @eldering proposes to use a separate lockfile and he will implement that.

Comment (nickygerritsen) [reply judge/judgedaemon.main.php:544]: Also I'd rather not change this variable name now, its pretty old.

Comment (meisterT) [reply judge/judgedaemon.main.php:544]: Please also log the path to the lockfile

Comment (nickygerritsen) [reply judge/judgedaemon.main.php:544]: @eldering will use a separate file and I guess he can add the logmsg for the filename 😄

**Review by nickygerritsen — COMMENTED** [review]
LGTM

**Review by eldering — APPROVED** [review]

**Pull request merged**